fidesctl/src/fidesctl/core/config/cli_settings.py

```diff
@@ -12,15 +13,33 @@ class CLISettings(FidesSettings):
     """Class used to store values from the 'cli' section of the config."""
 
     local_mode: bool = False
-    server_url: str = "http://localhost:8080"
+    # Ignored due to this bug: https://github.com/samuelcolvin/pydantic/issues/3143
+    server_host: AnyHttpUrl = "http://localhost"  # type: ignore
+    server_port: Optional[int]
+    server_url: Optional[AnyHttpUrl]
     analytics_id: str = generate_client_id(FIDESCTL_CLI)
 
+    @root_validator(skip_on_failure=True)
+    def get_database_url(cls: FidesSettings, values: Dict) -> Dict:
+        """
+        Create the server_url.
+
+        This needs to be set as a root validator,
+        otherwise the type errors gets suppressed.
+        """
+        server_host = values["server_host"]
+        server_port = values["server_port"]
+        values["server_url"] = "{}{}".format(
+            server_host,
+            ":" + str(server_port) if server_port else "",
+        )
+        return values
```

Comment:
Including the protocol in the `server_host` value makes the option name misleading, and defeats a lot of the purpose of decomposing the full URL. We should instead accept only a string hostname and a port number (and maybe a protocol, if we want to support multiple):

```suggestion
    analytics_id: str = generate_client_id(FIDESCTL_CLI)
    
    server_host: str = "localhost"
    server_port: Optional[int]
    server_protocol: str = "http"	# if you want, otherwise always http
    server_url: AnyHttpUrl

    @root_validator(skip_on_failure=True)
    def get_database_url(cls: FidesSettings, values: Dict) -> Dict:
        """
        Create the server_url.

        This needs to be set as a root validator,
        otherwise the type errors gets suppressed.
        """
        host = values["server_host"]
        port = values["server_port"]
        protocol = values["server_protocol"]

        values["server_url"] = "{}://{}{}".format(
        	protocol,
        	host,
            ":" + str(port) if port else "",
        )
        
        return values
```
My original goal with this work was to use `urllib.parse` to handle constructing the output, but after reading the docs and trying some things locally, it's clear that the library just wasn't intended for this. There doesn't seem to be a way to guarantee an RFC 3986 compliant string using the standard library (and I don't think it's worth adding another dependency).